### DISCUSSION TITLE
How to prevent all layouts from being injected into the html

### DISCUSSION THREAD

**Discussion by patrickspiegel** [nuxt/nuxt#19475] [Questions]
Hey,

I have a use case, where the same site should be displayed for multiple companies. Each company should have their own layout. The companies belong to the same group, but don't interact with each other but the sites are hosted on the same servers. 

I Have defined layouts for each company in layouts folder and I am using `<NuxtLayout>` in `app.vue`. I have defined a global middleware that sets the layout of the requested page.
```
export default defineNuxtPlugin((to) => {
    addRouteMiddleware('theme', (to) => {
        to.meta.layout = companyLayout
}})
```

When running the dev server only the company specific layout is being fetched. But when I try that out in build && preview mode then all layouts are getting prefetched. 

**What steps should I take to prevent all layouts from being injected into the html in Nuxt 3?**

**Comment by Baroshem** [comment]
Hey,

What about generating page content on the client side? I know that it is bad for performance but from what you say here this sounds to me a bit like each company can be refered to as a different `user`.

If you create a page/layout for each of these users, when you generate your app as SSG, all these pages/routes/layouts will be created as .html files so you won't be able to hide them.

It works in local dev environment as it is based on SSR (which will fetch create only the one page that you need at certain point).

So either remain on the SSR (so when a certain page is requested for company it will be created on the server and send to the client) or switch layouts to client only so that they will be populated with data after sending a client request for appriopriate data.

This is what I would do in this case :)
